README.md

```diff
@@ -60,17 +72,21 @@ class { 'servicenow_cmdb_integration':
 
-> If you're using a hiera-eyaml encrypted password, then make sure that `eyaml decrypt -s <encrypted_password>` returns the same password on all nodes in the `PE Master` node group.
+Commit the changes, then run Puppet on the node group. This will cause a restart of the `pe-puppetserver` service.
 
-> For customers using PE 2019.7+, we recommend adding this class to the `PE Master` node group. If HA is installed, then it should also be added to the `PE HA Replica` group.
+The `servicenow_cmdb_integration` class installs the `servicenow.rb` script and points Puppet’s `trusted_external_command` setting to that script. This script fetches the node's CMDB record from the `cmdb_ci` table and stores it in the `trusted.external.servicenow` variable.
 
-This class installs the `servicenow.rb` script. The script fetches the node's CMDB record from the `cmdb_ci` table and stores it in the `trusted.external.servicenow` variable. This variable is a hash of `<field> => <value>` where `<field>` is a field in the `cmdb_ci` table and `<value>` is the field's value. If the field's a reference field then `<value>` is that field's display value.
+You can set the `table` parameter to point to a different CMDB table (like e.g. `cmdb_ci_server`). However, keep in mind that the script will only fetch data for nodes that are in the specified table _or_ a child of that table (such as nodes in the `cmdb_ci_server` table or a child table like `cmdb_ci_server_hardware`). It will return an empty hash for all other nodes.
 
 **Note:** Puppet invokes the `servicenow.rb` script via the calling-convention `servicenow.rb <certname>`. The script fetches the node's CMDB record by querying the record whose `fqdn` field matches `<certname>`. If you're storing node certnames in a separate CMDB field, then you can set the `certname_field` parameter to that CMDB field's _column name_.
 
-**Note to readers from the classification section:** If you're already storing the node's environment and classes in your own CMDB fields, then make sure to specify those fields' column names in the `environment_field` and `classes_field` parameters, respectively. Otherwise, these default to `u_puppet_environment` and `u_puppet_classes`, respectively.
+**Note to readers who are enabling classification:** If you're already storing the node's environment and classes in their own CMDB fields, then make sure to specify those fields' _column names_ in the `environment_field` and `classes_field` parameters, respectively. Otherwise, these default to `u_puppet_environment` and `u_puppet_classes`, respectively.
+
+3. (Optional) To verify that everything worked, click on one of the nodes in the Matching nodes tab of the PE Master group and scroll down to the node’s trusted facts. You should see something like:
 
 #### ServiceNow node classification
```

Comment:
I think some of this will need to be updated. E.g. we'll need to add the note about name-value type, the `Puppetfile` commit should also specify `latest` as the version, etc. So, there's some new stuff added in doc that will need to be transported here, but it probably isn't much. I think you got it right that the majority of the backfill is the trusted external data section.

Reply:
Added the name-value section and latest tag back in.